api/models/graphite.go

```diff
@@ -0,0 +1,62 @@
+package models
+
+import (
+	"gopkg.in/raintank/schema.v1"
+	"sort"
+)
+
+type GraphiteRender struct {
+	MaxDataPoints uint32   `json:"maxDataPoints" form:"maxDataPoints"`
+	Targets       []string `json:"target" form:"target" binding:"required"`
+	From          string   `json:"from" form:"from"`
+	Until         string   `json:"until" form:"until"`
+	To            string   `json:"to" form:"to"`
+	Format        string   `json:"format" form:"format"`
+}
+
+type GraphiteFind struct {
+	Query  string `json:"query" form:"query" binding:"required"`
+	From   int64  `json:"from" form:"from"`
+	Until  int64  `json:"until" form:"until"`
+	Format string `json:"format" form:"format"`
+	Jsonp  string `json:"jsonp" form:"jsonp"`
+}
+
+type MetricsDelete struct {
+	Query string `json:"query" form:"query" binding:"required"`
+}
+
+type MetricDefs []schema.MetricDefinition
+
+func (defs MetricDefs) MarshalJSONFast(b []byte) ([]byte, error) {
+	seen := make(map[string]struct{})
+
+	names := make([]string, 0, len(defs))
+
+	for i := 0; i < len(defs); i++ {
+		_, ok := seen[defs[i].Name]
+		if !ok {
+			names = append(names, defs[i].Name)
+			seen[defs[i].Name] = struct{}{}
+		}
+	}
+	sort.Strings(names)
+	b = append(b, '[')
+	for _, name := range names {
+		b = append(b, '"')
+		b = append(b, name...)
+		b = append(b, `",`...)
+	}
+	if len(defs) != 0 {
+		b = b[:len(b)-1] // cut last comma
+	}
+	b = append(b, ']')
+	return b, nil
+}
+
+func (defs MetricDefs) MarshalJSON() ([]byte, error) {
+	var b []byte
+	var err error
+	b, err = defs.MarshalJSONFast(b)
+	return b, err
+}
```

Comment:
> Renaming our MarshalJSONFast to just MarshalJSON would prevent us from ever being able to support using json.Marashal() without causing backwards compatibility problems as Golang does not support method overloading. And that is a bad thing.

but MetricDefs is a type internal to metrictank, so if we ever make breaking changes to the api, then we can just adapt all calling code while we're at it.
I'm not a big fan of adding code just because we might need it in the future, unless you have concrete ideas of where we will need it?

BTW you could write the slow function like so:
```
func (defs MetricDefs) MarshalJSON() ([]byte, error) {
	return defs.MarshalJSONFast(nil)
}
```